XpringKit/RippledFlags.swift

```diff
@@ -0,0 +1,15 @@
+/// Flags used in rippled transactions.
+///
+/// These are only flags which are utilized in Xpring SDK. For a complete list of flags, see:
+/// https://xrpl.org/transaction-common-fields.html#flags-field.
+public struct RippledFlags: OptionSet {
+  public let rawValue: UInt32
+
+  /// Flag values.
+  public static let tfPartialPayment = RippledFlags(rawValue: 1 << 17)
```

Comment:
Is this just for detecting partial payments? SDK does not support sending partial payments, correct?

I'm actually curious why we _need_ to detect partial payments. Shouldn't looking at the `amount` field of the transaction be enough to determine how much we actually received?

Reply:
> Is this just for detecting partial payments? 
Correct

> SDK does not support sending partial payments, correct?
Correct

> I'm actually curious why we need to detect partial payments. 
We're trying to bucket transactions because we want to make the payments use case dead simple. 

> Shouldn't looking at the amount field of the transaction be enough to determine how much we actually received?

I guess we could do that. That seemed slightly more complicated than just checking for this flag and the marginal value felt pretty low.

Aside, and recognizing that this is a weak justification:  We need flags anyway (we should really be setting [tfCanonicalSig](https://xrpl.org/transaction-common-fields.html#global-flags) for instance so this seemed like an okay way to do this.

Reply:
BTW: This is how this class ultimately gets used: 
https://github.com/xpring-eng/Xpring-JS/pull/193

Reply:
So what happens if someone sends you a partial payment, and we're using `xpring-sdk`? We receive the transaction, say we can't parse it, but then when you fetch your account balance again it'll show it as incremented (from that transaction?)

Reply:
This API is for `getTransactionStatus`, so you give us a transaction hash and we give you an enum. It's maybe better to rename this method to `getPaymentStatus`. 

`getBalance` will just go straight to the ledger. 

There's no incoming hooks for this stuff; so you would need to BYOTxHash either by pulling it from TxHistory or by some other means.